test/src/opt-unit-libhdfs.cc

```diff
@@ -0,0 +1,189 @@
+#include <limits.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <sys/time.h>
+#include <sys/types.h>
+#include "catch.hpp"
+#include "hdfs.h"
+
+static off_t totalSize;
+static struct timeval tm1;
+
+struct LibHDFSFx {
+  static inline void start() {
+    gettimeofday(&tm1, NULL);
+  }
+
+  static inline void stop() {
+    struct timeval tm2;
+    gettimeofday(&tm2, NULL);
+
+    unsigned long long t =
+        1000000 * (tm2.tv_sec - tm1.tv_sec) + (tm2.tv_usec - tm1.tv_usec);
+    printf("Time: %llu us, MB/s:%f\n", t, (double)totalSize / (double)t);
+  }
+
+  LibHDFSFx() {
+  }
+
+  ~LibHDFSFx() {
+  }
+
+  int check_hdfs_write(
+      const char *writeFileName, off_t fileTotalSize, tSize bufferSize) {
+    hdfsFS fs;
+    hdfsFile writeFile;
+    char *buffer;
+    int i;
+    off_t nrRemaining;
+    tSize curSize;
+    tSize written;
+
+    fs = hdfsConnect("default", 0);
+    if (!fs) {
+      fprintf(stderr, "Oops! Failed to connect to hdfs!\n");
+      exit(-1);
```

Comment:
It would be good to test using the Catch test macros.  You can use the other tests for examples.